src/test/java/stepdefinitions/MyStepdefs.java

```diff
@@ -93,6 +96,84 @@ public void theUserSeesOneProductInShoppingCart(final String EXP_Q) throws Inter
         org.junit.Assert.assertEquals(EXP_Q, amountOfProduct);
     }
 
+    @When("Get product list")
+    public void getProductList() {
+        homePage.getList();
+    }
+
+    @And("Check quantity equals expected")
+    public void checkQuantityEqualsExpected() {
+        actualSizeOfProductList = homePage.getSizeOfProductList();
+        Assert.assertEquals(Constants.EXP_PRODUCT_LIST_SIZE, actualSizeOfProductList);
```

Comment:
Add an error message to All your Assertions, for example:
Assert.assertEquals(actualResult, expectedResult, String.format("Expected number of products '%s' but actual is '%s'",actualResult, expectedResult));